File: FunFinance.API/Models/Expense.cs
using System.ComponentModel.DataAnnotations;

namespace FunFinance.API.Models
{
    public class Expense
    {
        public int Id { get; set; }

        [Required, StringLength(100)]
        public required string Description { get; set; }

        [Required]
        public decimal Amount { get; set; }

        [Required]
        public DateTime Date { get; set; }

        // Зовнішні ключі
        public int CategoryId { get; set; }
        public int UserId { get; set; }
        public int BudgetId { get; set; }

        // Навігаційні властивості
        public required virtual Category Category { get; set; }
        public required virtual User User { get; set; }
        public required virtual Budget Budget { get; set; }
    }
}

File: funfinance-frontend/src/models/expense.ts
export interface Expense {
    id: number;
    description: string;
    amount: number;
    date: string;
    categoryId: number;
    userId: number;
    budgetId: number;
    category?: Category;
    user?: User;
    budget?: Budget;
  }
  
  export interface ExpenseCreate {
    description: string;
    amount: number;
    date: string;
    categoryId: number;
    userId: number;
    budgetId: number;
  }
  
  export interface ExpenseUpdate {
    description?: string;
    amount?: number;
    date?: string;
    categoryId?: number;
  }
  
  // Імпортуємо інші моделі
  import { Category } from './category';
  import { User } from './user';
  import { Budget } from './budget';